src/component/container/LayerTreeDropdownContextMenu/LayerTreeDropdownContextMenu.tsx

```diff
@@ -46,204 +43,188 @@ type ComponentProps = LayerTreeDropdownContextMenuDefaultProps & LayerTreeDropdo
-      isLoadingExtent: true
-    });
+  const changeMetadataModalVisibility = (): void => {
+    setMetadataModalVisible(!metadataModalVisible);
+  };
+
+  /**
+   * Tries to retrieve the layer extent from capabilities and fits map view to
+   * the bounds on success.
+   */
+  const zoomToLayerExtent = async (): Promise<void> => {
+
+    setExtentLoading(true);
 
     try {
       let extent = await LayerUtil.getExtentForLayer(layer, map.getView().getProjection());
-
       extent = transformExtent(extent, 'EPSG:4326', map.getView().getProjection());
-
       map.getView().fit(extent);
     } catch (error) {
       Logger.error(error);
       notification.error({
         message: t('LayerTreeDropdownContextMenu.extentError')
       });
     } finally {
-      this.setState({
-        isLoadingExtent: false,
-        menuHidden: true
-      });
+      setExtentLoading(false);
     }
   };
 
   /**
-   * The render function.
+   * Zooms map view to max layer resolution.
    */
-  render() {
-    const {
-      t,
-      layer,
-      showZoomToLayerExtent
-    } = this.props;
-
-    const {
-      infoModalVisible,
-      menuHidden,
-      metaDataModalVisible
-    } = this.state;
-
-    const showDesciption = !_isEmpty(layer.get('description'));
-    const showMetadata = !_isEmpty(layer.get('metadataIdentifier')) && layer.get('showMetadataInClient');
-
-    const settingsMenu = (
-      <Menu
-        selectable={false}
-        onClick={this.onContextMenuItemClick}
-      >
-        {
-          showDesciption &&
-          <MenuItem
-            key="info"
-          >
-            {t('LayerTreeDropdownContextMenu.layerInfoText')}
-          </MenuItem>
-        }
-        {
-          showMetadata &&
-          <MenuItem
-            key="metadata"
-          >
-            {t('LayerTreeDropdownContextMenu.layerMetadataText')}
-          </MenuItem>
-        }
-        {
-          showZoomToLayerExtent &&
-          <MenuItem
-            key="zoomToExtent"
-          >
-            <Spin
-              spinning={this.state.isLoadingExtent}
-            >
-              {t('LayerTreeDropdownContextMenu.layerZoomToExtent')}
-            </Spin>
-          </MenuItem>
-        }
-      </Menu>
-    );
-
-    return (
-      <div>
-        <Modal
-          className="info-modal"
-          visible={infoModalVisible}
-          footer={null}
-          title={layer.get('name') || t('Modal.LayerDescription.title')}
-          onCancel={() => this.changeInfoModalVisibility()}
+  const zoomToLayerResolution = (): void => {
+
+    const mapResolutions = map.getView().getResolutions();
+    let maxResolution = layer.getMaxResolution();
+
+    if (_isFinite(maxResolution) && maxResolution !== 0) {
+      const closestMapResolution = mapResolutions.reduce((prev, curr) =>
+        Math.abs(curr - maxResolution) < Math.abs(prev - maxResolution) ? curr : prev
+      );
+      const zoomToResolution = mapResolutions.indexOf(closestMapResolution) + 1;
```

Comment:
Do we need to add the level always or only in case of `-1`?

Reply:
Since the maxResolution is always [excluded](https://openlayers.org/en/latest/apidoc/module-ol_layer_Base-BaseLayer.html), we have to go to the next one to ensure, that layer will be visible after zoom.